src/OpenTK.Mathematics/Vector/Vector2i.cs

```diff
@@ -0,0 +1,603 @@
+        {
+            result.X = a.X + b.X;
+            result.Y = a.Y + b.Y;
+        }
+
+        /// <summary>
+        /// Subtract one Vector from another.
+        /// </summary>
+        /// <param name="a">First operand.</param>
+        /// <param name="b">Second operand.</param>
+        /// <returns>Result of subtraction.</returns>
+        public static Vector2i Subtract(Vector2i a, Vector2i b)
+        {
+            Subtract(ref a, ref b, out a);
+            return a;
+        }
+
+        /// <summary>
+        /// Subtract one Vector from another.
+        /// </summary>
+        /// <param name="a">First operand.</param>
+        /// <param name="b">Second operand.</param>
+        /// <param name="result">Result of subtraction.</param>
+        public static void Subtract(ref Vector2i a, ref Vector2i b, out Vector2i result)
+        {
+            result.X = a.X - b.X;
+            result.Y = a.Y - b.Y;
+        }
+
+        /// <summary>
+        /// Multiplies a vector by an integer scalar.
+        /// </summary>
+        /// <param name="vector">Left operand.</param>
+        /// <param name="scale">Right operand.</param>
+        /// <returns>Result of the operation.</returns>
+        public static Vector2i Multiply(Vector2i vector, int scale)
+        {
+            Multiply(ref vector, scale, out vector);
+            return vector;
+        }
+
+        /// <summary>
+        /// Multiplies a vector by an integer scalar.
+        /// </summary>
+        /// <param name="vector">Left operand.</param>
+        /// <param name="scale">Right operand.</param>
+        /// <param name="result">Result of the operation.</param>
+        public static void Multiply(ref Vector2i vector, int scale, out Vector2i result)
+        {
+            result.X = vector.X * scale;
+            result.Y = vector.Y * scale;
+        }
+
+        /// <summary>
+        /// Multiplies a vector by the components a vector (scale).
+        /// </summary>
+        /// <param name="vector">Left operand.</param>
+        /// <param name="scale">Right operand.</param>
+        /// <returns>Result of the operation.</returns>
+        public static Vector2i Multiply(Vector2i vector, Vector2i scale)
+        {
+            Multiply(ref vector, ref scale, out vector);
+            return vector;
+        }
+
+        /// <summary>
+        /// Multiplies a vector by the components of a vector (scale).
+        /// </summary>
+        /// <param name="vector">Left operand.</param>
+        /// <param name="scale">Right operand.</param>
+        /// <param name="result">Result of the operation.</param>
+        public static void Multiply(ref Vector2i vector, ref Vector2i scale, out Vector2i result)
+        {
+            result.X = vector.X * scale.X;
+            result.Y = vector.Y * scale.Y;
+        }
+
+        /// <summary>
+        /// Divides a vector by a scalar.
```

Comment:
Indicate that it's performing integer division, floor(a/b).

same for the others below.